atlasdb-tests-shared/src/main/java/com/palantir/atlasdb/transaction/impl/AbstractTransactionTest.java

```diff
@@ -1309,6 +1318,165 @@ public void testGetRangesPaging() {
         verifyAllGetRangesImplsNumRanges(t, ranges, ImmutableList.of("v"));
     }
 
+    @Test
+    public void getRowsAccessibleThroughCopies() {
+        Transaction t = startTransaction();
+        byte[] rowKey = row(0);
+        byte[] value = value(0);
+        t.put(TEST_TABLE, ImmutableMap.of(
+                Cell.create(rowKey, column(0)), value));
+        t.commit();
+
+        t = startTransaction();
+        SortedMap<byte[], RowResult<byte[]>> result =
+                t.getRows(TEST_TABLE, ImmutableList.of(rowKey), ColumnSelection.all());
+        assertThat(result.get(rowKey))
+                .as("it should be possible to get a row from getRows with a passed-in byte array")
+                .isNotNull()
+                .satisfies(rowResult ->
+                        assertThat(rowResult.getOnlyColumnValue()).isEqualTo(value));
+
+        byte[] rowKeyCopy = rowKey.clone();
+        assertThat(rowKeyCopy).isNotSameAs(rowKey);
+        assertThat(result.get(rowKeyCopy))
+                .as("it should be possible to get a row from getRows with a copy of a passed-in byte array")
+                .isNotNull()
+                .satisfies(rowResult ->
+                        assertThat(rowResult.getOnlyColumnValue()).isEqualTo(value));
+    }
+
+    @Test
+    public void getRowsSortedByByteOrder() {
+        Transaction t = startTransaction();
+        byte[] row0 = row(0);
+        byte[] row1 = row(1);
+        byte[] col0 = column(0);
+        t.put(TEST_TABLE, ImmutableMap.of(Cell.create(row0, col0), value(0), Cell.create(row1, col0), value(1)));
+        t.commit();
+
+        t = startTransaction();
+        SortedMap<byte[], RowResult<byte[]>> readRows =
+                t.getRows(TEST_TABLE, ImmutableList.of(row0, row1), ColumnSelection.all());
```

Comment:
Do we test somewhere the ColumnSelection and ordering of columns?

Reply:
Not that I'm aware of, I added one test.